frontend/packages/dev-console/src/components/import/ImportStrategySection.tsx

```diff
@@ -0,0 +1,162 @@
+import * as React from 'react';
+import { AlertVariant, Alert, ButtonVariant, Button } from '@patternfly/react-core';
+import { PencilAltIcon, UndoIcon } from '@patternfly/react-icons';
+import { FormikValues, useFormikContext } from 'formik';
+import { useTranslation } from 'react-i18next';
+import { ImportStrategy } from '@console/git-service/src';
+import { PipelineKind } from '@console/pipelines-plugin/src/types';
+import { NormalizedBuilderImages } from '../../utils/imagestream-utils';
+import BuilderSection from './builder/BuilderSection';
+import DevfileStrategySection from './devfile/DevfileStrategySection';
+import DockerSection from './git/DockerSection';
+import ImportStrategySelector from './ImportStrategySelector';
+import FormSection from './section/FormSection';
+
+export interface ImportStrategySectionProps {
+  builderImages: NormalizedBuilderImages;
+  existingPipeline?: PipelineKind;
+}
+
+const ImportStrategySection: React.FC<ImportStrategySectionProps> = ({ builderImages }) => {
+  const { t } = useTranslation();
+  const { values, setFieldValue, setValues } = useFormikContext<FormikValues>();
+  const {
+    import: {
+      strategies: importStrategies,
+      loaded,
+      loadError,
+      selectedStrategy,
+      showEditImportStrategy,
+      recommendedStrategy,
+    },
+    devfile,
+    docker,
+  } = values;
+  const [alertVariant, setAlertVariant] = React.useState<AlertVariant>(AlertVariant.default);
+  const [alertTitle, setAlertTitle] = React.useState<string>('');
+  const [alertDescription, setAlertDescription] = React.useState<React.ReactNode>('');
+  const [importStrategyComponent, setImportStrategyComponent] = React.useState<React.ReactNode>(
+    null,
+  );
+  const [recommendedValues, setRecommendedValues] = React.useState<any>(null);
+
+  React.useEffect(() => {
```

Comment:
I think instead of useEffect and useState here, a useMemo can work here to get the Alert props and component to render